torch_np/_ndarray.py

```diff
@@ -436,19 +439,57 @@ def sum(self, axis=None, dtype=None, out=None, keepdims=NoValue,
         if initial is not None or where is not None:
             raise NotImplementedError
 
-        if dtype is None:
-            dtype = self.dtype
-        if _dtypes.is_integer(dtype):
-            dtype = _dtypes.default_float_type()
-        torch_dtype = _dtypes.torch_dtype_from(dtype)
-
+        torch_dtype = _helpers.float_or_default(dtype, self.dtype)
         if axis is None:
             result = self._tensor.sum(dtype=torch_dtype)
         else:
             result = self._tensor.sum(dtype=torch_dtype, dim=axis)
 
         return result
 
+    @axis_out_keepdims_wrapper
+    def prod(self, axis=None, dtype=None, out=None, keepdims=NoValue,
+            initial=NoValue, where=NoValue):
+        if initial is not None or where is not None:
+            raise NotImplementedError
+
+        axis = _helpers.allow_only_single_axis(axis)
+
+        torch_dtype = _helpers.float_or_default(dtype, self.dtype)
+        if axis is None:
+            result = self._tensor.prod(dtype=torch_dtype)
+        else:
+            result = self._tensor.prod(dtype=torch_dtype, dim=axis)
+
+        return result
```

Comment:
Nit. You can golf these sort of patterns as
```suggestion
kwargs = {"dim": axis} if axis is not None else {}
return self._tensor.prod(dtype=torch_dtype, **kwargs)
```